components/cam/src/dynamics/se/restart_dynamics.F90

```diff
@@ -115,7 +115,7 @@ subroutine write_restart_dynamics(File, dyn_out)
     type(element_t), pointer :: elem(:)
     real(kind=r8) :: time
     integer :: ndcur, nscur
-    integer, pointer :: ldof(:)
+    integer(kind=pio_offset_kind), pointer :: ldof(:)
```

Comment:
How does this help? Is this for bigger numbers?

Reply:
Yeah, we ran into integer overflows with ne1024. Specifying the larger integer type prevents that.